uber/site_sections/registration.py

```diff
@@ -500,7 +500,7 @@ def register(self, session, message='', **params):
                 message = 'Please select a payment type'
             elif attendee.payment_method == c.MANUAL and not re.match(c.EMAIL_RE, attendee.email):
                 message = 'Email address is required to pay with a credit card at our registration desk'
-            elif attendee.badge_type not in [c.ATTENDEE_BADGE, c.ONE_DAY_BADGE]:
+            elif attendee.badge_type not in [badge for badge, desc in c.AT_THE_DOOR_BADGE_OPTS]:
```

Comment:
This allows for the presale, but doesn't enforce the constraint of only being able to check in using badges on these days.  We also need to modify the ``Attendee.can_check_in`` property so that if someone shows up and tries to claim their ``Thursday`` badge on ``Saturday`` it won't let them without an admin changing the badge type and asking them for the difference in cost.  That should be like a one-line change, I'd expect.